plugin-server/src/worker/ingestion/timestamps.ts

```diff
@@ -119,3 +119,30 @@ export function parseDate(supposedIsoString: string): DateTime {
     }
     return DateTime.fromJSDate(jsDate).toUTC()
 }
+
+export function toYearMonthDayInTimezone(
+    timestamp: number,
+    timeZone: string
+): { year: number; month: number; day: number } {
+    const parts = new Intl.DateTimeFormat('en', {
+        timeZone,
+        year: 'numeric',
+        month: '2-digit',
+        day: '2-digit',
+    }).formatToParts(new Date(timestamp))
+    const year = parts.find((part) => part.type === 'year')?.value
+    const month = parts.find((part) => part.type === 'month')?.value
+    const day = parts.find((part) => part.type === 'day')?.value
+    if (!year || !month || !day) {
+        throw new Error('Failed to get year, month, or day')
+    }
```

Comment:
I don't believe this can ever happen, `formatToParts` will always return a value. If you attempt to pass anything that couldn't be formatted it will raise `RangeError`

![image](https://github.com/user-attachments/assets/bdaabf46-2238-4f9b-b61f-104f0df19376)

Are you doing this for TS purposes? If so, because `formatToParts` never fails to return a value we might do this to collect the values instead - it's deterministic AFAICT

```typescript
const [month, , day, , year] = parts
```

Reply:
I assume this is just to make typescript happy?

Reply:
Yeah it was - I looked through the docs and I do think the order is guaranteed https://tc39.es/ecma402/#sec-formatdatetimetoparts but I'm not confident enough to assume this